index.bs

```diff
@@ -10361,19 +10363,12 @@ is the [=interface=]:
         [=overload resolution algorithm=].
     1.  Let |R| be the result of performing the actions listed in the description of
         |constructor| with |values| as the argument values.
+    1.  Assert: |R| is an object that implements |I|.
```

Comment:
R is something that's not concretely defined since we don't have requirements for what IDL-prose can and cannot be. I agree that it cannot really be a completion value, but I also agree with @domenic that it can have thrown an exception (constructors do this) and we do need to handle that somehow.